pkg/apis/core/v1/resource_test.go

```diff
@@ -10,9 +11,42 @@ import (
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	gomegaTypes "github.com/onsi/gomega/types"
 )
 
 var _ = Describe("resource.Resource", func() {
+	Context("ResourceWithTags", func() {
+		rwt := &ResourceWithTag{Identifier: "test-identifier", Tag: "test-tag"}
+
+		DescribeTable("Test EndpointURL and FilterRequestURL for all operations", func(op types.Operation, errorMatcher gomegaTypes.GomegaMatcher, expectedPath string) {
+			singleObjectOperation := op == types.OperationGet || op == types.OperationUpdate || op == types.OperationDestroy
+			ctxWithOperation := types.ContextWithOperation(
+				context.TODO(),
+				op,
+			)
+
+			url, err := rwt.EndpointURL(ctxWithOperation)
+			Expect(err).To(errorMatcher)
+
+			if err == nil {
+				Expect(url.Path).To(BeEquivalentTo(expectedPath))
+				// API client appends objects identifier to path on singleObjectOperation which should be removed by FilterRequestURLHook
+				if singleObjectOperation {
+					url.Path = path.Join(url.Path, rwt.Identifier)
+				}
+				filteredURL, err := rwt.FilterRequestURL(ctxWithOperation, url)
+				Expect(err).To(errorMatcher)
+				Expect(filteredURL.Path).To(BeEquivalentTo(expectedPath))
+			}
+		},
+			Entry("When operation is Create", types.OperationCreate, BeNil(), "/api/core/v1/resource.json/test-identifier/tags/test-tag"),
```

Comment:
Uuuuh, passing a `Matcher` instead of just an `error` is way more elegant than my previous go-to method of doing that - nice job!